pallets/gear-program/src/migration.rs

```diff
@@ -209,3 +209,117 @@ mod test {
+                log::error!("❌ Migration is not allowed for current storage version {current:?}.");
+                return weight;
+            }
+
+            let update_to = StorageVersion::new(MIGRATE_TO_VERSION);
+            log::info!("🚚 Running migration from {onchain:?} to {update_to:?}, current storage version is {current:?}.");
+            let mut counter = 0;
+            let mut removal_result =
+                paused_program_storage::PausedProgramStorage::<T>::clear(u32::MAX, None);
+            // MultiRemovalResults contains two fields on which we calculate weight:
+            // - loops: how many iterations of loop were performed, each requiring read
+            // - backend: number of elements removed from database, corresponds to a write.
+
+            weight = weight.saturating_add(
+                T::DbWeight::get()
+                    .reads_writes(removal_result.loops as u64, removal_result.backend as u64),
+            );
+            counter += removal_result.backend;
+
+            while let Some(cursor) = removal_result.maybe_cursor.take() {
+                removal_result = paused_program_storage::PausedProgramStorage::<T>::clear(
+                    u32::MAX,
+                    Some(&cursor),
+                );
+                weight = weight.saturating_add(
+                    T::DbWeight::get()
+                        .reads_writes(removal_result.loops as u64, removal_result.backend as u64),
+                );
+                counter += removal_result.backend;
+            }
+
+            // Put new storage version
+            weight = weight.saturating_add(T::DbWeight::get().writes(1));
+
+            update_to.put::<Pallet<T>>();
+
+            log::info!("✅ Successfully migrated storage. {counter} codes have been migrated");
+        } else {
+            log::info!("🟠 Migration requires onchain version {MIGRATE_FROM_VERSION}, so was skipped for {onchain:?}");
+        }
+
+        weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+        const MIGRATE_FROM_VERSION: u16 = 5;
+        let current = Pallet::<T>::current_storage_version();
+        let onchain = Pallet::<T>::on_chain_storage_version();
+
+        let res = if onchain == MIGRATE_FROM_VERSION {
+            ensure!(
+                current == ALLOWED_CURRENT_STORAGE_VERSION,
+                "Current storage version is not allowed for migration, check migration code in order to allow it."
+            );
+
+            Some(0)
+        } else {
+            None
+        };
+
+        Ok(res.encode())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(_state: Vec<u8>) -> Result<(), TryRuntimeError> {
+        ensure!(
+            paused_program_storage::PausedProgramStorage::<T>::iter().count() == 0,
+            "Paused program storage is not empty after upgrade"
+        );
+        Ok(())
+    }
+}
+
+mod paused_program_storage {
```

Comment:
You can impl pallets/gear-program/src/migrations/* in different files
```suggestion
mod onchain {
```